repository/src/main/java/org/alfresco/repo/jscript/AlfrescoScriptThreadMxBeanWrapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2022 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software. 
+ * If the software was purchased under a paid Alfresco license, the terms of 
+ * the paid license agreement will prevail.  Otherwise, the software is 
+ * provided under the following open source license terms:
+ * 
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.jscript;
+
+import java.lang.management.ManagementFactory;
+
+/**
+ * Allows to monitor memory usage
+ */
+public class AlfrescoScriptThreadMxBeanWrapper
+{
+    protected com.sun.management.ThreadMXBean threadMXBean; // Only Oracle / OpenJDK
```

Comment:
Why can't we use `ManagementFactory` from the `java.lang` package? I mean `java.lang.management.ManagementFactory.getThreadMXBean();` We should not really write codes that call 'sun' packages.

Reply:
I had a quick look and I don't think we can use `ThreadMXBean` from `java.lang` to get the allocated memory. However, the current implementation needs to be hardened as not all the JVM implementations support this which causes an exceptions to be thrown.
So first, we need to check whether the JVM supports thread memory allocation or not, then if it is enabled.

Reply:
I am not sure on my reading of the interface but I think @tiagosalvado10's approach maybe the right now.  I do see there are two methods that could be used to test if 1/it is supported ([isThreadAllocatedMemorySupported](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.management/com/sun/management/ThreadMXBean.html#isThreadAllocatedMemorySupported())) and then 2/ if it is enabled ([isThreadAllocatedMemoryEnabled()](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.management/com/sun/management/ThreadMXBean.html#isThreadAllocatedMemoryEnabled())). However the class documentation states:

> Platform-specific management interface for the thread system of the Java virtual machine.
This platform extension is only available to a thread implementation that supports this extension.

So if it is not supported on the platform will the interface not be available? I think this might be something we need to verify if we can do the tests or will be get the ClassNotFoundException that is already being handled.

Reply:
I have delegated the checks to `AlfrescoScriptThreadMxBeanWrapper` class, handling the exceptions in case of interface not being available and also using the methods mentioned by Jared to determine if memory check is supported. 

I have performed tests against OpenJDK and Corretto and it has worked on both.